### ISSUE TITLE
[template/conditional-complexity]: Not reporting failures in some cases

### ISSUE THREAD

**Issue by rafaelss95** [angular-eslint/angular-eslint#280]
~**Case 1**: When using other bound attribute than `ngIf`:~ (Fixed by #279).

```html
<div [class.px-4]="a <= b || (b > c && c >= d && d < e)">
  Content
</div>
```
<br>

~**Case 2**: When the condition is within a `Interpolation`:~ (Fixed by #279).

```html
{{ control && control.invalid && (control.touched || (showOnDirty && control.dirty)) ? 'some value' : 'another value' }}
```
<br>

**Case 3**: When the condition contains a `Pipe` (that isn't the last statement), it doesn't report anything, no matter how complex it is.

```html
<ng-container *ngIf="(control | isDefined) && control.invalid && (control.touched || (showOnDirty && control.dirty))">
  Content
</ng-container>
```
<br>

~**Case 4**: The rule only doesn't "watch" `Conditional` even in bound attributes:~ (Fixed by #279).

```html
<div [class.px-4]="a === 'x' ? v === c : b === 3 ? 0 : c">Test</div>
```

Relates to:

- https://github.com/mgechev/codelyzer/issues/627
- https://github.com/mgechev/codelyzer/issues/630

**Comment by drakenfly** [comment]
I found a related issue. The following example of the no-negated-async pipe should lead to an error in the false expression branch of a conditional (note: the pipe operator '|' binds stronger than the conditional 'condition ? trueExp : falseExp'):
```{{ boolVar ? 'success' : !(obsVar | async) }}````.

This problem arises due to the Conditional AST being not traversed further.

```
const KEYS: VisitorKeys = {
  ASTWithSource: ['ast'],
  Binary: ['left', 'right'],
  BoundAttribute: ['value'],
  BoundEvent: ['handler'],
  BoundText: ['value'],
  Conditional: ['condition', 'falseExp', 'trueExp'], // <-- add me !
  Element: ['children', 'inputs', 'outputs', 'attributes'],
  Interpolation: ['expressions'],
  PrefixNot: ['expression'],
  Program: ['templateNodes'],
  PropertyRead: ['receiver'],
  Template: ['templateAttrs', 'children', 'inputs'],
};
```

**Comment by rafaelss95** [comment]
@JamesHenry I just tested the missing case (3rd scenario in OP) in latest (v12) and it seems to work fine now. The same is valid for `{{ boolVar ? 'success' : !(obsVar | async) }}` posted in the comment above. What do you think about adding tests for these cases and then close this issue?
